src/activities/quizzes/attempts/QuizAttemptsEntity.js

```diff
@@ -106,6 +127,32 @@ export class QuizAttemptsEntity extends Entity {
 		return this._entity && this._entity.getSubEntityByClass(Classes.quizzes.attempts.retakeIncorrectOnly);
 	}
 
+	/**
+	 * @returns {object} quiz attempt conditions sub-entity
+	 */
+	getAttemptConditionsSubEntity() {
+		if (this._entity.hasClass(Classes.quizzes.attempts.attemptConditions) && this._entity.hasClass('collection')) {

```

Comment:
Constant for 'collection'?

Reply:
We are checking that the current entity has the `attempt-conditions` class but also that the sub-entity has the same class name?
Does both the entity and sub-entity use the same class name?

Probably also a bit overkill to check both. Just getting the sub-entity is probably sufficient.

Reply:
Also if the entity is a `collection`, shouldn't you want to `getSubentities` instead of just 1 entity?

Reply:
>Does both the entity and sub-entity use the same class name?
Different by just an `s` in their `class`.

@ronan-f We don't have a constant for `collection` in `hypermedia-constants`
@wongvincent Should we make a constant for it?